WordPress/src/main/java/org/wordpress/android/ui/mysite/quickstart/QuickStartBlockBuilder.kt

```diff
@@ -1,39 +1,36 @@
 package org.wordpress.android.ui.mysite.quickstart
 
 import org.wordpress.android.R
-import org.wordpress.android.fluxc.store.QuickStartStore
 import org.wordpress.android.fluxc.store.QuickStartStore.QuickStartTaskType
 import org.wordpress.android.ui.mysite.MySiteItem.QuickStartBlock
 import org.wordpress.android.ui.mysite.MySiteItem.QuickStartBlock.QuickStartTaskTypeItem
-import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.ui.mysite.QuickStartRepository.QuickStartCategory
 import org.wordpress.android.ui.utils.ListItemInteraction
 import org.wordpress.android.ui.utils.UiString.UiStringRes
 import org.wordpress.android.ui.utils.UiString.UiStringResWithParams
 import org.wordpress.android.ui.utils.UiString.UiStringText
 import javax.inject.Inject
 
-class QuickStartBlockBuilder
-@Inject constructor(
-    private val appPrefsWrapper: AppPrefsWrapper,
-    private val quickStartStore: QuickStartStore
-) {
-    fun build(onItemClick: (QuickStartTaskType) -> Unit): QuickStartBlock {
-        val localSiteId = appPrefsWrapper.getSelectedSite().toLong()
-
+class QuickStartBlockBuilder @Inject constructor() {
+    fun build(
+        categories: List<QuickStartCategory>,
+        onItemClick: (QuickStartTaskType) -> Unit
+    ): QuickStartBlock {
         val taskTypeItems = mutableListOf<QuickStartTaskTypeItem>()
-        taskTypeItems.add(buildQuickStartTaskTypeItem(localSiteId, QuickStartTaskType.CUSTOMIZE, onItemClick))
-        taskTypeItems.add(buildQuickStartTaskTypeItem(localSiteId, QuickStartTaskType.GROW, onItemClick))
+        categories.forEach { category ->
+            taskTypeItems.add(buildQuickStartTaskTypeItem(category, onItemClick))
+        }
 
         return QuickStartBlock(taskTypeItems = taskTypeItems)
     }
 
     private fun buildQuickStartTaskTypeItem(
-        localSiteId: Long,
-        quickStartTaskType: QuickStartTaskType,
+        category: QuickStartCategory,
         onItemClick: (QuickStartTaskType) -> Unit
     ): QuickStartTaskTypeItem {
-        val countCompleted = quickStartStore.getCompletedTasksByType(localSiteId, quickStartTaskType).size
-        val countUncompleted = quickStartStore.getUncompletedTasksByType(localSiteId, quickStartTaskType).size
+        val quickStartTaskType = category.taskType
+        val countCompleted = category.completedTasks.size
+        val countUncompleted = category.uncompletedTasks.size
```

Comment:
Question (❓): So, calling `quickStartStore` is not required anymore and is removed as being redundant right? And that is done because we already have the number of completed/uncompleted tasks from the `categories` flowing from the source(s), right? I just wonder why it was implemented like this in the first place... 🤔

Reply:
> So, calling quickStartStore is not required anymore and is removed as being redundant right? And that is done because we already have the number of completed/uncompleted tasks from the categories flowing from the source(s), right? 

Right

>  I just wonder why it was implemented like this in the first place...

The logic to get completed/ uncompleted tasks count got [migrated from the `MySiteFragment` which uses `quickStartStore`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt#L650-L665)   in [this PR](https://github.com/wordpress-mobile/WordPress-Android/issues/15127) focussing only on displaying the quick start section and not on updating it. 

With the focus on refreshing the quick start section items, this PR took this further and hooked the repository's categories (holding completed/uncompleted tasks) so that the list items can be updated using the `PartialState` logic used in `ImprovedMySiteFragment`. It also felt more natural to use the repository/ source to access the `quickStartStore` rather than the builder accessing it directy.

Reply:
👋 @ashiagr !

Thank you so much for the great explanation, it all makes sense to me now, great work! 🙇  🚀